akka-stream/src/main/scala/akka/stream/io/IOResult.scala

```diff
@@ -0,0 +1,33 @@
+package akka.stream.io
+
+import akka.Done
+import scala.util.{ Failure, Success, Try }
+
+/**
+ * Holds a result of an IO operation.
+ *
+ * @param count A number of IO operations performed.
```

Comment:
Yeah, I think this should always talk about bytes written / read.
It should be final too OR not a case class which is also interesting... Let's go with final though, others can introduce new types if they need to.